lib/cloud_controller/runners/puma_runner.rb

```diff
@@ -0,0 +1,51 @@
+require 'puma'
+require 'puma/configuration'
+require 'puma/events'
+
+module VCAP::CloudController
+  class PumaRunner
+    def initialize(config, app, logger, periodic_updater, request_logs)
+      @logger = logger
+      @periodic_updater = periodic_updater
+      @request_logs = request_logs
+      puma_config = Puma::Configuration.new do |conf|
+        # this is actually called everytime a worker is started
+        # https://github.com/puma/puma/blob/5f3f489ee867317c47724d0fc5b1d906f1b23de6/lib/puma/dsl.rb#L607
+        # we probably want to come up with a different way to do this.  Perhaps a singleton?
+        conf.after_worker_fork {
+          Thread.new do
+            EM.run do
+              @periodic_updater.setup_updates
+            end
+          end
+        }
```

Comment:
Why do we still need to do this? I thought switching from statsd to Prometheus meant we don't need to periodically emit metrics?

Reply:
Neither Prometheus nor statsd style metrics are going to work well with Puma clusters yet.  A lot of the current metrics aren't set up to consider multiple processes.  

Prometheus metrics are going to require us to figure out a caching system that the processes will share if we want to emit metrics via a Prometheus endpoint, since each process tracks metrics and you can't guarantee which process will respond to the prometheus query.

The statsd_injector acts as a shared memory store, so we thought that it might be a better option for now, but after some testing, we found there is no simple way of migrating our existing metrics to a format that works for multiple processes.

Our current thinking is to either use the imperfect statsd metrics for now, or just remove the metrics entirely until we are able to do a thorough rethinking of what metrics will be valuable for a Puma cluster deployment of CCNG--some of the metrics we have right now were designed to help operators scale CCNG on Thin, for example, and aren't necessarily useful for a Puma deployment.